### ISSUE TITLE
Implement post commit hook for automatically triggering an Atlas build

### ISSUE THREAD

**Issue by ptwobrussell** [MakerPress/atlas-public-feedback#284]
It would be convenient if I could somehow trigger a post commit hook to automatically trigger a (configurable) new build so that the build starts immediately when I commit the code (similar to how the old system with the arcane orm:commit or whatever it was used to work with svn commits.)  For example, if I included "build:pdf build:html" in my commit message, then perhaps Atlas could build my pdf and html version of my book for me automatically. I realize that this isn't necessary by any means and would be a power user luxury since it's not that big of a deal to login to Atlas and trigger a build, but this seems like a legitimate use of a post commit hook, so I wanted to request it as a nice to have feature. Does any such thing already exist?

Not to conflate the issue with something else, but what would also be nice is if I would get an email with a link to automatically download my built-artifacts such as the PDF once the build is done, or better yet, offer me a web hook that would get pinged with this same information so that when the build is ready, I could automatically take some action with the generated artifacts (such as download them to a server that I control.) I have more use for the post commit hook that the post-build notification at the moment, but if you offered both, I think it would open up some creative possibilities.

**Comment by runemadsen** [comment]
Thanks @ptwobrussell. This is actually going to be possible in the new version that we're working on. It's going to be much more API-driven, and both of the things you mention would be possible with a tiny bit of client side scripting.
